src/three/plugins/ImplicitTilingPlugin.js

```diff
@@ -0,0 +1,56 @@
+import {SUBTREELoader} from "./SUBTREELoader.js";
+
+export class ImplicitTilingPlugin {
+
+	constructor() {
+		this.name = 'IMPLICIT_TILING_PLUGIN';
+	}
+
+	init(tiles) {
+
+		this.tiles = tiles;
+
+	}
+
+	preprocessNode(tile, uri) {
+		if (tile.implicitTiling) {	//only for root
```

Comment:
When we say "root" - does this mean the root of the tileset? Or the root of an implicit subtree, which could be 5 tiles deep, for example?

Reply:
It's for the root of the tileset :+1:

Reply:
Perhaps I'm misunderstanding but from reading the spec it looks like this is _not_ only for the root of the tile set. It's a bit confusing but the spec [makes a distinction](https://github.com/CesiumGS/3d-tiles/tree/main/specification/ImplicitTiling#implicit-root-tile) between the root of the tile set and the root of an implicit subtree. Note that it says the "implicitTiling" object can be added to any tile in the tile set:

> An implicitTiling object may be added to ***any tile*** in the tileset JSON. Such a tile is called an implicit root tile, to distinguish it from the root tile of the tileset JSON.

Reply:
Yes you're right, a child declaring implicitTiling object should meet this condition. The comment is misleading. As I said before, I didn't test the case where a child is declaring an implicitTiling object. I've only implemented the most straightforward configuration, e.g with the implicitTiling object in the root of the tileset.

Reply:
Okay I see now - I was under the impression that there was a separate of the spec that handled child subtrees differently that wasn't implemented. Since this is implemented generically the logic for parsing shouldn't be any different - especially since we're using __level now instead of __depth. We should just pass the implicit tiling root in, instead. The associated implicit root can be found by either climbing the tree until the implicit root is found or by saving a reference to it on all child tiles.

Reply:
I'm now storing the Implicit Root Tile as `__implicitRoot`and it's carried by each children. It means the Implicit Root Tile is storing itself (circular reference).

Reply:
Great! Regarding https://github.com/NASA-AMMOS/3DTilesRendererJS/issues/608#issuecomment-2283364305 - does it make sense to change this line item:

> ImplicitTiling defined in children. Only the one in root is supported

to

> Test implicit tiling defined in children

It _should_ be supported since we've implemented what's needed against the spec. It just hasn't been tested.